idunn/blocks/wikipedia.py

```diff
@@ -118,6 +148,69 @@ def wrapper(*args, **kwargs):
 
         return wrapper
 
+class WikipediaCache:
+    _expire = None
+    _connection = None
+
+    @classmethod
+    def init_cache(cls):
+        from app import settings
+
+        redis_url = settings['WIKI_API_REDIS_URL']
+
+        if redis_url is not None:
+            cls._expire = int(settings['WIKI_CACHE_TIMEOUT']) * 3600 # seconds
+
+            ip, port = redis_url.split(":")
+            redis_db = settings['WIKI_CACHE_REDIS_DB']
+            redis_timeout = int(settings['WIKI_REDIS_TIMEOUT'])
+
+            try:
+                cls._connection = Redis(
+                    connection_pool=RedisConnection(host=ip, port=port, db=redis_db, timeout=redis_timeout).pool()
+                )
+            except RedisNoConnException:
+                logging.warning("No Redis instance available for caching", exc_info=True)
+                cls._connection = False
+        else:
+            logging.warning("No Redis URL has been set for caching", exc_info=True)
+            cls._connection = False
+
+    @classmethod
+    def cache_it(cls, prefix, f):
+        """
+        Takes function f and put its result in a redis cache.
+        It requires a prefix string to identify the name
+        of the function cached.
+        """
+        if WikipediaCache._connection is None:
+            WikipediaCache.init_cache()
+
+        def with_cache(*args, **kwargs):
+            """
+            If the redis is up we use the cache,
+            otherwise we bypass it
+            """
+            if cls._connection is not False:
+                """
+                The key is the function name (prefix)
+                along with the args.
+                """
+                args_list = [arg for ak in zip(args, kwargs.values()) for arg in ak]
```

Comment:
This seems broken to me.
How zip is supposed to work here ? And `self` should be removed from args.  
We probably don't need this complexity. Why not passing a complete key to `cache_it` that could be used in redis directly ?